nats.go

```diff
@@ -3635,13 +3673,38 @@ func (s *Subscription) handleControlMessage(msg *Msg) (bool, error) {
 		if err != nil {
 			return isControl, err
 		}
+	} else {
+		// Process heartbeat received, get latest control metadata if present.
+		var ctrl *controlMetadata
+		cmeta := jsi.cmeta.Load()
+		if cmeta == nil {
+			return isControl, nil
+		}
+
+		ctrl = cmeta.(*controlMetadata)
+		tokens, err := getMetadataFields(ctrl.meta)
```

Comment:
Is this from last message we received successfully?

Reply:
Yes these would be from the raw metadata of a previous message.